### ISSUE TITLE
go panic: interface conversion with v0.9.0-alpha release

### ISSUE THREAD

**Issue by mgtrrz** [segmentio/terraform-docs#203]
### Prerequisites

Put an `x` into the box that applies:

- [x] This issue describes a bug.
- [ ] This issue describes a feature request.

For more information, see the [Contributing Guidelines](https://github.com/segmentio/terraform-docs/tree/master/CONTRIBUTING.md).

### Description

terraform-docs crashes in latest terraform-docs update

#### Actual Behavior

Running `terraform-docs markdown table ./` on a module we have for rds_postgres.

```
panic: interface conversion: interface {} is int, not float64

goroutine 1 [running]:
github.com/segmentio/terraform-docs/internal/types.ValueOf(0x135ee60, 0x1736aa0, 0x135ee60, 0x1736aa0)
	/Users/mgutierrez/go/src/github.com/segmentio/terraform-docs/internal/types/types.go:38 +0x57d
github.com/segmentio/terraform-docs/internal/module.loadInputs(0xc0000e2070, 0xd, 0x0, 0x0, 0xcc00123b4c0, 0x0, 0x0, 0xc0000b52c0, 0xc0000b52e0, 0xc0000b5320)
	/Users/mgutierrez/go/src/github.com/segmentio/terraform-docs/internal/module/module.go:97 +0x2cb
github.com/segmentio/terraform-docs/internal/module.LoadWithOptions(0xc0001128a0, 0xc000112c90, 0xc0001128a0, 0x0)
	/Users/mgutierrez/go/src/github.com/segmentio/terraform-docs/internal/module/module.go:25 +0xa1
github.com/segmentio/terraform-docs/cmd.doPrint(0x7ffeefbff536, 0xd, 0x14985c0, 0xc0000a26c0, 0x0, 0x0)
	/Users/mgutierrez/go/src/github.com/segmentio/terraform-docs/cmd/root.go:86 +0xc4
github.com/segmentio/terraform-docs/cmd.glob..func5(0x1713e00, 0xc000099f30, 0x1, 0x1, 0x0, 0x0)
	/Users/mgutierrez/go/src/github.com/segmentio/terraform-docs/cmd/markdown.go:24 +0x67
github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra.(*Command).execute(0x1713e00, 0xc000099ef0, 0x1, 0x1, 0x1713e00, 0xc000099ef0)
	/Users/mgutierrez/go/src/github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra/command.go:840 +0x460
github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x17145e0, 0x0, 0x8a000, 0xc000088058)
	/Users/mgutierrez/go/src/github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra/command.go:945 +0x317
github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/Users/mgutierrez/go/src/github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra/command.go:885
github.com/segmentio/terraform-docs/cmd.Execute(...)
	/Users/mgutierrez/go/src/github.com/segmentio/terraform-docs/cmd/root.go:62
main.main()
	/Users/mgutierrez/go/src/github.com/segmentio/terraform-docs/main.go:10 +0x2e
```

Possibly introduced from: https://github.com/segmentio/terraform-docs/pull/196 ?

#### Expected Behavior

The command should generate docs as expected. This command ran correctly on previous versions of terraform-docs.

### Replication Case

Tell us which steps to take to replicate your problem.

1. Create a terraform module (or probably any terraform file with a variable that's using a number/integer type

```
variable "create_read_replicas" {
  description = "If set to 1, create a read replica for each primary node"
  default     = 0
}
```

It also fails with numbers not set to 0:

```
variable "example_delay" {
 description = "Determines how long (in milliseconds) the standby server should wait"

 default = 1800000
}
```
2. Run `terraform-docs markdown table <directory>`
3. Go panic

### Version

```
$ terraform-docs --version
terraform-docs version v0.9.0-alpha darwin/amd64 BuildDate: 2020-02-20T10:13:35-06:00
```

**Comment by khos2ow** [comment]
Thanks for the report, trying to reproduce it now.

**Comment by khos2ow** [comment]
I've tried with both of your provided examples and strangely neither failed. In #196 I specifically added these exactly to test this case: 

https://github.com/segmentio/terraform-docs/blob/27b95c4adf4d13ccc6f7619c07c705e4aecabad0/examples/variables.tf#L31-L49

The only other think I can think of at the moment is the Go version. Which version of go you tried this with?

**Comment by mgtrrz** [comment]
Apologies, the variables with int values was the first thing i commented out that seemed to work but it seems to be the existence of both a variable with a number type and a variable with a variable in the description. Try this:

```
variable "application" {
    default = "test"
}

variable "test_int" {
    default = 10
}

variable "db_name" {
  description = "Name of main/default database. Defaults to ${var.application} (stripped of any '-' characters)"
  default     = ""
}
```

**Comment by apriljo** [comment]
Same issue with using a variable in description of an output.

`❯ cat april/main.tf`
```
variable "number" {
  description = "the number"
  default = 42
}

output "module_number" {
  description = "the module number, defaults ${var.number}"
  value = "${var.number}"
}
```

`❯ terraform-docs markdown table april`
```
panic: interface conversion: interface {} is int, not float64

goroutine 1 [running]:
github.com/segmentio/terraform-docs/internal/types.ValueOf(0x135ee40, 0xc00009d1d8, 0x135ee40, 0xc00009d1d8)
	/Users/abaird/go/src/github.com/segmentio/terraform-docs/internal/types/types.go:38 +0x57d
github.com/segmentio/terraform-docs/internal/module.loadInputs(0xc0001302a0, 0x5, 0x0, 0x0, 0xcc0011b34c0, 0x0, 0x0, 0xc0000ad2c0, 0xc0000ad2e0, 0xc0000ad320)
	/Users/abaird/go/src/github.com/segmentio/terraform-docs/internal/module/module.go:97 +0x2cb
github.com/segmentio/terraform-docs/internal/module.LoadWithOptions(0xc00010a8a0, 0xc00010ac90, 0xc00010a8a0, 0x0)
	/Users/abaird/go/src/github.com/segmentio/terraform-docs/internal/module/module.go:25 +0xa1
github.com/segmentio/terraform-docs/cmd.doPrint(0x7ffeefbff9ae, 0x5, 0x14985a0, 0xc0000986c0, 0x0, 0x0)
	/Users/abaird/go/src/github.com/segmentio/terraform-docs/cmd/root.go:86 +0xc4
github.com/segmentio/terraform-docs/cmd.glob..func5(0x1713e00, 0xc00008ff30, 0x1, 0x1, 0x0, 0x0)
	/Users/abaird/go/src/github.com/segmentio/terraform-docs/cmd/markdown.go:24 +0x67
github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra.(*Command).execute(0x1713e00, 0xc00008fef0, 0x1, 0x1, 0x1713e00, 0xc00008fef0)
	/Users/abaird/go/src/github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra/command.go:840 +0x460
github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x17145e0, 0x0, 0x76400, 0xc000084058)
	/Users/abaird/go/src/github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra/command.go:945 +0x317
github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/Users/abaird/go/src/github.com/segmentio/terraform-docs/vendor/github.com/spf13/cobra/command.go:885
github.com/segmentio/terraform-docs/cmd.Execute(...)
	/Users/abaird/go/src/github.com/segmentio/terraform-docs/cmd/root.go:62
main.main()
	/Users/abaird/go/src/github.com/segmentio/terraform-docs/main.go:10 +0x2e
```

If I use terraform-docs at commit 38e1897 it works.

**Comment by khos2ow** [comment]
Really weird, I'm still not able to reproduce the number issue, but `db_name` with variable inside the description failed for me too. Can you provide these details just in case:

- `uname`
- `go version`
- `terraform version`
- `make build && ./bin/<YOUR_PLATFORM>-amd64/terraform-docs --version`

**Comment by apriljo** [comment]
```
❯ uname -a
Darwin X-0015 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020; root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64
❯ go version
go version go1.13.7 darwin/amd64
❯ terraform version
Terraform v0.11.11

Your version of Terraform is out of date! The latest version
is 0.12.21. You can update by downloading from www.terraform.io/downloads.html
❯ terraform-docs version
terraform-docs version v0.9.0-alpha darwin/amd64 BuildDate: 2020-02-20T12:11:22-05:00
```
Currently on latest master terraform-docs installed by `go get -u github.com/segmentio/terraform-docs`

**Comment by mgtrrz** [comment]
```
$ uname
Darwin
$ go version
go version go1.13.4 darwin/amd64
$ terraform version
Terraform v0.11.14

~/go/src/github.com/segmentio/terraform-docs > master $ make build
==> Clean workspace
rm -rf ./bin
==> Build binary for current OS/ARCH
CGO_ENABLED=0 GO111MODULE=on go build -mod=vendor -ldflags " -X github.com/segmentio/terraform-docs/internal/version.version=v0.8.2-17-g27b95c4 -X github.com/segmentio/terraform-docs/internal/version.commitHash=27b95c4 -X github.com/segmentio/terraform-docs/internal/version.buildDate=2020-02-20T11:12:38-0600 " -o ./bin/darwin-amd64/terraform-docs
~/go/src/github.com/segmentio/terraform-docs > master $ ./bin/darwin-amd64/terraform-docs --version
terraform-docs version v0.8.2-17-g27b95c4 darwin/amd64 BuildDate: 2020-02-20T11:12:38-0600

~/tmptf $ ls test_module 
variables.tf

~/tmptf $ cat test_module/variables.tf 
variable "application" {
    default = "test"
}

variable "test_int" {
    default = 10
}

variable "db_name" {
  description = "Name of main/default database. Defaults to ${var.application} (stripped of any '-' characters)"
  default     = ""
}

~/tmptf/ $ ~/go/src/github.com/segmentio/terraform-docs/bin/darwin-amd64/terraform-docs markdown table test_module
panic: interface conversion: interface {} is int, not float64
```

Similarly, this happens in CircleCI, where terraform-docs is used and where we originally noticed the error. That version of go is `circleci/golang:1.12.4`

**Comment by khos2ow** [comment]
Thank you so much folks for giving the detailed response, the only difference I can see here is that both of you are on MacOS and I'm on Linux (which would be really weird if it was the case). I'm gonna try to find a Mac somewhere and try the same thing over there to rule that out.

My attempts up to now have been as following in which the only issue I could reproduce was using the variable in the description:

- on `HEAD` of master
- on 38e1897 
- with updated `terraform-config-inspect` (both standalone binary and the vendor inside `terraform-docs`)

For the sake of argument, would you be able to use this instead of line 38 in `internal/types/types.go` and try again? (the good ol' console log! :man_facepalming:)

```go
	case "int", "int8", "int16", "int32", "int64", "float32", "float64":
		switch i := v.(type) {
		case float64:
			fmt.Println("float64")
			return Number(float64(i))
		case float32:
			fmt.Println("float32")
			return Number(float64(i))
		case int64:
			fmt.Println("int64")
			return Number(float64(i))
		case int32:
			fmt.Println("int32")
			return Number(float64(i))
		case int16:
			fmt.Println("int16")
			return Number(float64(i))
		case int8:
			fmt.Println("int8")
			return Number(float64(i))
		case int:
			fmt.Println("int")
			return Number(float64(i))
		}
```

**Comment by apriljo** [comment]
That works for me.
```
❯ cd $GOPATH/src/github.com/segmentio/terraform-docs
❯ git pull
Already up to date.
❯ vi internal/types/types.go
❯ git diff origin/master
diff --git a/internal/types/types.go b/internal/types/types.go
index 2302788..476a7de 100644
--- a/internal/types/types.go
+++ b/internal/types/types.go
@@ -35,7 +35,29 @@ func ValueOf(v interface{}) Default {
                }
                return String(v.(string))
        case "int", "int8", "int16", "int32", "int64", "float32", "float64":
-               return Number(v.(float64))
+               switch i := v.(type) {
+               case float64:
+                       fmt.Println("float64")
+                       return Number(float64(i))
+               case float32:
+                       fmt.Println("float32")
+                       return Number(float64(i))
+               case int64:
+                       fmt.Println("int64")
+                       return Number(float64(i))
+               case int32:
+                       fmt.Println("int32")
+                       return Number(float64(i))
+               case int16:
+                       fmt.Println("int16")
+                       return Number(float64(i))
+               case int8:
+                       fmt.Println("int8")
+                       return Number(float64(i))
+               case int:
+                       fmt.Println("int")
+                       return Number(float64(i))
+               }
        case "bool":
                return Bool(v.(bool))
        case "[]interface {}":
❯ make build && ./bin/darwin-amd64/terraform-docs --version
==> Clean workspace
rm -rf ./bin
==> Build binary for current OS/ARCH
CGO_ENABLED=0 GO111MODULE=on go build -mod=vendor -ldflags " -X github.com/segmentio/terraform-docs/internal/version.version=v0.8.2-17-g27b95c4 -X github.com/segmentio/terraform-docs/internal/version.commitHash=27b95c4 -X github.com/segmentio/terraform-docs/internal/version.buildDate=2020-02-20T12:54:44-0500 " -o ./bin/darwin-amd64/terraform-docs
terraform-docs version v0.8.2-17-g27b95c4 darwin/amd64 BuildDate: 2020-02-20T12:54:44-0500
❯ ./bin/darwin-amd64/terraform-docs markdown table april
int
## Providers

No provider.

## Inputs

| Name | Description | Type | Default | Required |
|------|-------------|------|---------|:--------:|
| number | the number | `number` | `42` | no |

## Outputs

| Name | Description |
|------|-------------|
| module\_number | the module number, defaults ${var.number} |
```

**Comment by apriljo** [comment]
I'm able to use that code change to compile the other docs that were failing on me too. 👍

**Comment by khos2ow** [comment]
That's interesting, in all my tests I'm getting `float64` all the time, but it seems you're getting `int`. I'm definitely gonna test this on a mac before pushing a PR.

**Comment by khos2ow** [comment]
This should be fixed in f55fd6cc5bcf72b463e04fa5b3110a7918b15370, please reopen the issue if you still encountered this.
